pkg/gui/types/version_number.go

```diff
@@ -0,0 +1,41 @@
+package types
+
+import (
+	"errors"
+	"regexp"
+	"strconv"
+)
+
+type VersionNumber struct {
```

Comment:
This has some overlap with `git_commands.GitVersion`, but that one has slightly different requirements, and I didn't find it trivial to unify the two.

Reply:
Agreed, from my own efforts. Actually, it wasn't hard to create a `SemanticVersion` that lifted the common stuff... but go's type system is not particularly ergonomic for what amount to "newtype" like scenarios, AFAICT. Either literals were pointlessly ugly or it was wrappers all over the place, all for DRYing up a few tests.